src/Common/Extensions/StringExtensions.cs

```diff
@@ -17,6 +17,11 @@ public static class StringExtensions
 
         private static readonly string UrlUserInfoReplacement = "***";
 
+        public static bool EqualsIgnoreCase(string a, string b)
```

Comment:
Instead of `StringExtensions.EqualsIgnoreCase(a, b)`, we can support `a.EqualsIgnoreCase(b)`...it is more concise

Also a subtle thing to note here: `a.Equals(b)` throws NRE if `a` is null, but we need not check if `a` is null in our extension case and delegate that responsibility to `string.Equals` method which correctly handles the equality here

Reply:
but then we'd need to check for `null` before using it.. the way it's implemented now, it behaves exactly like `string.Equals`

Reply:
I mean we need _not_ check for null. the current usage feels verbose and not I was originally expecting. @gpcastro what do you think?